Update bazis manual (#186)
ubvu/open-handbook: topics/bazis.qmd

### Connecting with SSH

Use your favourite SSH client to login at \<VUNETID\>@bazis.labs.vu.nl. On windows we recommend MobaXterm; Mac users can use iterm. Direct access is only possible from the Campus or from SURF. From other network locations first connect to the stepstone \<VUNETID\>@ssh.data.vu.nl, or use eduVPN Institutional Access (below). Here be provide the steps to connect to the cluster using MobaXterm.

1. Download and install MobaXTerm from [mobatek.net](https://mobaxterm.mobatek.net/download.html).
2. Open MobaXTerm and click on Session
3. Enter the Basic SSH settings as follows:
   1. Remote host: bazis.labs.vu.nl
   2. Tick "specify username"
   3. Enter your username (VUnetID)
4. Go to “Network settings” and click SSH gateway (jump host)
5. Enter the following details and click OK

>    1. Gateway host: ssh.data.vu.nl

Reviewer: ```suggestion
   1. Gateway host: `ssh.data.vu.nl`
```

   2. Enter your username (VUnetID)
6. Save the settings. When not connected automatically, you can click on User sessions, and double click "bazis.labs.vu.nl (VUNETID)"
7. You will be asked for a password for "ssh.data.vu.nl". This is your NORMAL VUNETID password, NOT the cluster password your received from ITVO.
8. If not connected automatically, connect to the server again. Now you will see a terminal window which asks for another password. Here you should enter the CLUSTER password that you received from ITVO. Note that you will not see the cursor moving when typing. This is normal. When you finished typing your password hit ENTER.
9. You will now see a terminal window, similar to this one. On connecting for the first time, you will be asked to change your CLUSTER password. Enter your CLUSTER password, then enter a new password (this can be identical to your VUNETID password).
10. Congratulations, you are now connected to the cluster! The next steps are to set up your scripts and data!

> As an alternative, you can use eduVPN to connect to the VU network, however the above configuration is recommended and generally works best in the long term. You can find the client [here](https://www.eduvpn.org/client-apps/). Start eduVPN and choose Institute Access to connect. You will need MFA to authenticate. Students can activate MFA at the servicedesk ([kb-item 11809](https://services.vu.nl/kb_view.do?sysparm_article=KB0011809)).

Reviewer: ```suggestion
As an alternative, you can use eduVPN to connect to the VU network, however the above configuration is recommended and generally works best in the long term. You can [find the client on the eduVPN pages](https://www.eduvpn.org/client-apps/). Start eduVPN and choose Institute Access to connect. You will need to enable Multi-Factor Authentication (MFA). Students can activate MFA at the servicedesk ([kb-item 11809](https://services.vu.nl/kb_view.do?sysparm_article=KB0011809)).
```

1. Right click on the file explorer of MobaXTerm on the left, and click "New empty file".
2. Call the file test.py and click OK.
3. Double click the new file in the file explorer, after which the MobaTextEditor will open. In the file you can write `print("hello world!")`.
4. Save the file, click "Yes" to sync the file to the server, and wait for it to synchronize.
5. In the terminal type `python test.py`.
6. You should see the output `hello world!` printed to the console.

Your first script was just executed on the login "node". A node can be thought of as a single computer of the cluster. This login node should only be used for doing very minor things, and any command you execute only runs while you are connected to the server. In the next steps you will learn how to execute a so-called "job" on one of the other nodes using slurm. A job can execute in the background, and if configured correctly, you will receive an email when the job is done.

> 1. Create a new file called `test.sh` (just like you did before), and enter the following in the file. Make sure to replace \<YOUR-EMAIL\>@vu.nl with your own email. When you copy the contents of this file on Windows you may need to set the "format" to Unix, which can be done using the top menu of the MobaXtermEditor (format -> UNIX).

Reviewer: ```suggestion
1. Create a new file called `test.sh` (just like you did before), and enter the following in the file. Make sure to replace `<YOUR-EMAIL>@vu.nl` with your own email. When you copy the contents of this file on Windows you may need to set the "format" to Unix, which can be done using the top menu of the MobaXtermEditor (format -> UNIX).
```

	f.write("hello world!")
```

You can also all or a selection of files from the scratch directory to your home directory after the job is done. This can be done by including the `cp` command in your slurm-file. For example, to copy all files from the scratch directory to your home directory, you can use the following command:

```bash
cp $TMPDIR/* ~/
```

## Data recovery

Most departments have a backup system in place, but configurations may vary. For example, an automatic backup may be made each day with a rentention period of 7 days, and a weekly backuup with a retention period of 5 weeks.

You can find all thour data under `.zfs/snapshot`. You can then copy your data from there to your home folder. 

```bash
cd ~/.zfs/snapshot
ls
```

Note that the folder is hidden, so you need to use `ls -a` to see it, or configure your file browser to show hidden files.

## Data deletion

You can only access the cluster with you VUNETID and thus when you are no longer affiliated with the VU, your account will be disabled. However, it is good practice to clean up your data when you are done with it. Depending on the department, data may be deleted after a certain period.

### Best practices

> - Make use of fast scratch directory ($TMPDIR).

Reviewer: ```suggestion
- Make use of fast scratch directory (`$TMPDIR`).
```

> - Don't run large computation on the login nodes! It negatively impacts all cluster users. Grab a compute node with srun --pty bash option.

Reviewer: ```suggestion
- Don't run large computation on the login nodes! It negatively impacts all cluster users. Grab a compute node with `srun --pty` bash option.
```

* [mathworks-parpool](https://nl.mathworks.com/help/parallel-computing/parpool.html)

## Acknowledging the cluster

When you use the cluster for your research, please acknowledge the cluster in your publications. You can use the following text:

>    > We kindly acknowledge sponsoring by the VU HPC Council and the IT for Research (ITvO) BAZIS Linux computational cluster at the VU University Amsterdam.

Reviewer: ```suggestion
   > We kindly acknowledge sponsoring by the VU HPC Council and the IT for Research (ITvO) BAZIS Linux computational cluster at VU Amsterdam.
```

Otherwise it reads Vrije University University Amsterdam 🤔
